File: Java/src/Encapsulamiento/CajaDeAhorro.java
package Encapsulamiento;


class CajaDeAhorro {

    private String titular;
    private double saldo;

    /**
     * post: la instancia queda asignada al titular indicado y con saldo igual a 0.
     */
    public CajaDeAhorro(String titularDeLaCuenta) {
        this.titular = titularDeLaCuenta;
        this.saldo = 0;
    }

    /**
     * post: devuelve el nombre del titular de la Caja de Ahorro.
     */
    public String obtenerTitular() {
        return titular;
    }

    /**
     * post: devuelve el saldo de la Caja de Ahorro.
     */
    public double consultarSaldo() {
        return saldo;
    }

    /**
     * pre : monto es un valor mayor a 0.
     * post: aumenta el saldo de la Caja de Ahorro según el monto depositado.
     */
    public void depositar(double monto) {
        if (monto > 0) {
            saldo += monto;
        } else {
            System.out.println("No se puede depositar un monto negativo o 0");
        }
    }

    /**
     * pre : monto es mayor a 0 y menor o igual que el saldo de la Caja de Ahorro.
     * post: disminuye el saldo de la Caja de Ahorro según el monto extraído.
     */
    public void extraer(double monto) {
        if (monto > 0 && monto <= this.saldo) {
            saldo -= monto;
        } else {
            System.out.println("No se puede extraer un monto negativo o mayor que el saldo");
        }
    }
}

File: Python/Encapsulamiento/caja-de-ahorro.py
class Caja_de_ahorro:


    def __init__(self, titular) -> None:
        self.__titular = titular
        self.__saldo = 0.0
    
    def obtener_titular(self) -> str:
        return self.__titular
    
    def consultar_saldo(self) -> float:
        return self.__saldo
    
    def depositar(self, monto):
        if monto > 0:
            self.__saldo += monto
        else:
            print("El monto a depositar debe ser mayor a cero")
    
    def extraer(self, monto):
        if monto > 0 and monto <= self.__saldo:
            self.__saldo -= monto
        else:
            print("El monto a extraer debe ser mayor a cero y no superar el saldo en la cuenta")
    
